cts/cts-cli.in

```diff
@@ -1,3727 +1,3409 @@
+# but probably something we want to continue warning about elsewhere
+# pylint: disable=invalid-name
+# pacemaker imports need to come after we modify sys.path, which pylint will complain about.
+# pylint: disable=wrong-import-position
+# We know this is a very long file.
+# pylint: disable=too-many-lines
+
+__copyright__ = "Copyright 2024 the Pacemaker project contributors"
+__license__ = "GNU General Public License version 2 or later (GPLv2+) WITHOUT ANY WARRANTY"
+
+import argparse
+from contextlib import contextmanager
+from datetime import datetime, timedelta
+import fileinput
+from functools import partial
+from gettext import ngettext
+from multiprocessing import Pool, cpu_count
+import os
+import pathlib
+import re
+from shutil import copyfile
+import signal
+from string import Formatter
+import subprocess
+import sys
+from tempfile import NamedTemporaryFile, TemporaryDirectory, mkstemp
+import types
+
+# These imports allow running from a source checkout after running `make`.
+if os.path.exists("@abs_top_srcdir@/python"):
+    sys.path.insert(0, "@abs_top_srcdir@/python")
+
+# pylint: disable=comparison-of-constants,comparison-with-itself,condition-evals-to-constant
+if os.path.exists("@abs_top_builddir@/python") and "@abs_top_builddir@" != "@abs_top_srcdir@":
+    sys.path.insert(0, "@abs_top_builddir@/python")
+
+from pacemaker._cts.errors import XmlValidationError
+from pacemaker._cts.validate import validate
+from pacemaker.buildoptions import BuildOptions
+from pacemaker.exitstatus import ExitStatus
+
+# Individual tool tests are split out, but can also be accessed as a group with "tools"
+tools_tests = ["cibadmin", "crm_attribute", "crm_standby", "crm_resource",
+               "crm_ticket", "crmadmin", "crm_shadow", "crm_verify"]
+# The default list of tests to run, in the order they should be run
+default_tests = ["access_render", "daemons", "dates", "error_codes"] + tools_tests + \
+                ["crm_mon", "acls", "validity", "upgrade", "rules", "feature_set"]
+other_tests = ["agents"]
+
+# The directory containing this program
+test_home = os.path.dirname(os.path.realpath(__file__))
+
+# The name of the shadow CIB
+SHADOW_NAME = "cts-cli"
+
+# Arguments to pass to valgrind
+VALGRIND_ARGS = ["-q", "--gen-suppressions=all", "--show-reachable=no", "--leak-check=full",
+                 "--trace-children=no", "--time-stamp=yes", "--num-callers=20",
+                 "--suppressions=%s/valgrind-pcmk.suppressions" % test_home]
+
+
+class PluralFormatter(Formatter):
+    """
+    Special string formatting class for selecting singular vs. plurals.
+
+    Use like so:
+    fmt = PluralFormatter()
+    print(fmt.format("{0} {0}:plural,test,tests} succeeded", n_tests))
+    """
+
+    def format_field(self, value, format_spec):
+        """Convert a value to a formatted representation."""
+        if format_spec.startswith("plural,"):
+            _, singular, plural = format_spec.split(',')
```

Comment:
would be nice if plural could default to singular + s but not imporant

Reply:
Easy enough to add.